hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/cache/EntityMapWarmerTest.java

```diff
@@ -0,0 +1,639 @@
+                .build();
+        return Transaction.newBuilder()
+                .setBodyBytes(ByteString.copyFrom(cryptoUpdateTxnBody.toByteArray()))
+                .build()
+                .toByteArray();
+    }
+
+    private byte[] newTransferTxn(List<TokenTransferList> tokenTransfers) {
+        final var now = newNowTimestamp();
+        final var memo = newTxnMemo(now);
+        var cryptoTransferBody = CryptoTransferTransactionBody.newBuilder()
+                .setTransfers(TransferList.newBuilder()
+                        .addAccountAmounts(AccountAmount.newBuilder()
+                                .setAccountID(HBAR_TRANSFER_ACCOUNT_1)
+                                .setAmount(-1)
+                                .build())
+                        .build())
+                .addAllTokenTransfers(tokenTransfers)
+                .build();
+        final var cryptoTransferTxnBody = TransactionBody.newBuilder()
+                .setCryptoTransfer(cryptoTransferBody)
+                .setMemo(memo)
+                .setTransactionID(
+                        TransactionID.newBuilder().setTransactionValidStart(now).build())
+                .build();
+        return Transaction.newBuilder()
+                .setBodyBytes(ByteString.copyFrom(cryptoTransferTxnBody.toByteArray()))
+                .build()
+                .toByteArray();
+    }
+
+    private Round mockRoundWithTxn() {
+        return mockRoundWithTxns(new SwirldTransaction(newTransferTxn()));
+    }
+
+    private Round mockRoundWithTxns(SwirldTransaction... txns) {
+        final List<com.swirlds.common.system.transaction.Transaction> roundTxns = List.of(txns);
+
+        var event = mock(EventImpl.class);
+        Mockito.lenient().when(event.transactionIterator()).thenReturn(roundTxns.iterator());
+        lenient().doCallRealMethod().when(event).forEachTransaction(notNull());
+
+        final var round = mock(ConsensusRound.class);
+        lenient().when(round.getRoundNum()).thenReturn(1L);
+        lenient()
+                .when(round.iterator())
+                .thenReturn(List.of((ConsensusEvent) event).iterator());
+        return round;
+    }
+
+    private void verifyNoCacheWarming() {
+        verify(accountAdpt, never()).getOnDiskAccounts();
+        verify(nftAdpt, never()).getOnDiskNfts();
+        verify(tokenRelAdpt, never()).getOnDiskRels();
+    }
+
+    private static Timestamp newNowTimestamp() {
+        return Timestamp.newBuilder()
+                .setSeconds(System.currentTimeMillis() / 1000)
+                .setNanos(1)
+                .build();
+    }
+
+    private static class RecursiveSynchronousThreadpoolDouble extends ThreadPoolExecutor {
+        private final List<Runnable> taskQueue = new ArrayList<>();
+
+        RecursiveSynchronousThreadpoolDouble() {
+            this(Mockito.mock(BlockingQueue.class));
+        }
+
+        RecursiveSynchronousThreadpoolDouble(BlockingQueue<Runnable> workQueue) {
```

Comment:
This seems like a hack to avoid using java's fork/join support.  Might be worth re-examining in that light at some point (particularly if this pattern is repeated outside of test code, or used in many places in test code).

Reply:
It is indeed very hacky. When I wrote the map warmer the primary focus by far was getting a working prototype to experiment with. As far as I know the timeline requirement hasn't changed, and this code is needed/preferred/wanted/?? in production (hence the designation of 'proto-duction' in the class javadoc 😅). Happy to rework the entire approach if the requirement isn't as pressing as I thought though, or at some point in the future.